### PR TITLE
Fix serverController get method and add test case to check state overwrite

### PR DESCRIPTION
**Pull request by psypersky**

### CODE REVIEW

**Comment by unity-haiwei** [comment]
Please merge this ASAP. it's breaking our build....

File: src/index.js [hunk 1]
```diff
@@ -112,10 +112,9 @@ module.exports.ServerController = function (state) {
     accessors: {
       get: function (path) {
         path = typeof path === 'string' ? path.split('.') : path
-        while (path.length) {
-          state = state[path.shift()]
-        }
-        return state
+        return path.reduce(function (currentState, key) {
```
Comment (unity-haiwei) [inline src/index.js]: Cannot read property 'setting' of undefined, when the path is 'head.setting'. In fact, simply no ‘head’.

Comment (Guria) [inline src/index.js]: can you provide test case?

Comment (unity-haiwei) [inline src/index.js]: var ctrl = ServerController({foo: null, 'boo': 'bee'}) from server render.

Comment (Guria) [inline src/index.js]: Cc @christianalfoni

Comment (christianalfoni) [inline src/index.js]: Hm... yeah, I think you are right @unity-haiwei. It should return "undefined". So:

```js
  return path.reduce(function (currentState, key) {
      return currentState === undefined || currentState === null ? undefined : currentState[key]
   }, state)
```

Now it returns undefined if the path does not exist or you try to access a key on `null`.

**Pull request merged**